<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Processor</class>
 <widget class="QDockWidget" name="Processor">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>400</width>
    <height>520</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>400</width>
    <height>520</height>
   </size>
  </property>
  <property name="features">
   <set>QDockWidget::DockWidgetFloatable|QDockWidget::DockWidgetMovable</set>
  </property>
  <property name="windowTitle">
   <string>Processor</string>
  </property>
  <widget class="QWidget" name="dockWidgetContents">
   <layout class="QGridLayout" name="gridLayout_2" columnstretch="20,40,40">
    <item row="0" column="2" rowspan="8">
     <widget class="QGroupBox" name="groupBox">
      <property name="title">
       <string>Registers</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="Register" name="R0"/>
       </item>
       <item>
        <widget class="Register" name="R1"/>
       </item>
       <item>
        <widget class="Register" name="R2"/>
       </item>
       <item>
        <widget class="Register" name="R3"/>
       </item>
       <item>
        <widget class="Register" name="R4"/>
       </item>
       <item>
        <widget class="Register" name="R5"/>
       </item>
       <item>
        <widget class="Register" name="R6"/>
       </item>
       <item>
        <widget class="Register" name="R7"/>
       </item>
       <item>
        <widget class="Register" name="R8"/>
       </item>
       <item>
        <widget class="Register" name="R9"/>
       </item>
       <item>
        <widget class="Register" name="RA"/>
       </item>
       <item>
        <widget class="Register" name="RB"/>
       </item>
       <item>
        <widget class="Register" name="RC"/>
       </item>
       <item>
        <widget class="Register" name="RD"/>
       </item>
       <item>
        <widget class="Register" name="RE"/>
       </item>
       <item>
        <widget class="Register" name="RF"/>
       </item>
      </layout>
     </widget>
    </item>
    <item row="7" column="0" colspan="2">
     <layout class="QGridLayout" name="gridLayout">
      <item row="2" column="1">
       <widget class="QPushButton" name="ButtonStep">
        <property name="text">
         <string>Step</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QPushButton" name="ButtonRunStop">
        <property name="text">
         <string>Run</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QPushButton" name="ButtonReset">
        <property name="text">
         <string>Reset</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QPushButton" name="ButtonLoad">
        <property name="text">
         <string>Load</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0" colspan="2">
       <widget class="QSlider" name="ClockSpeed">
        <property name="toolTip">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Processor speed&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="maximum">
         <number>10</number>
        </property>
        <property name="singleStep">
         <number>1</number>
        </property>
        <property name="pageStep">
         <number>2</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="invertedAppearance">
         <bool>false</bool>
        </property>
        <property name="tickPosition">
         <enum>QSlider::TicksBothSides</enum>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="4" column="0" colspan="2">
     <spacer name="verticalSpacer">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>20</width>
        <height>40</height>
       </size>
      </property>
     </spacer>
    </item>
    <item row="6" column="0" colspan="2">
     <widget class="QGroupBox" name="groupBox_4">
      <property name="title">
       <string>Accumulator</string>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout" stretch="40,60">
       <item>
        <widget class="Register" name="DF"/>
       </item>
       <item>
        <widget class="Register" name="D"/>
       </item>
      </layout>
     </widget>
    </item>
    <item row="0" column="0">
     <widget class="QGroupBox" name="groupBox_2">
      <property name="title">
       <string>Flags</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="ExternalFlag" name="EF1">
         <property name="text">
          <string>EF1</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="ExternalFlag" name="EF2">
         <property name="text">
          <string>EF2</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="ExternalFlag" name="EF3">
         <property name="text">
          <string>EF3</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="ExternalFlag" name="EF4">
         <property name="text">
          <string>EF4</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QGroupBox" name="groupBox_3">
      <property name="title">
       <string>Control</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <widget class="Register" name="X"/>
       </item>
       <item>
        <widget class="Register" name="P"/>
       </item>
       <item>
        <widget class="Register" name="T"/>
       </item>
       <item>
        <widget class="Register" name="I"/>
       </item>
       <item>
        <widget class="Register" name="N"/>
       </item>
      </layout>
     </widget>
    </item>
    <item row="3" column="1">
     <widget class="IndicatorLED" name="IE">
      <property name="text">
       <string>IE</string>
      </property>
     </widget>
    </item>
    <item row="1" column="1">
     <widget class="IndicatorLED" name="Q">
      <property name="text">
       <string>Q</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Register</class>
   <extends>QLineEdit</extends>
   <header>register.h</header>
  </customwidget>
  <customwidget>
   <class>ExternalFlag</class>
   <extends>QCheckBox</extends>
   <header>externalflag.h</header>
  </customwidget>
  <customwidget>
   <class>IndicatorLED</class>
   <extends>QLabel</extends>
   <header>indicatorled.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
